application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationServiceImpl.java

```diff
@@ -421,6 +441,16 @@ private TraineeRecommendationRecordDto buildTraineeRecommendationRecordDto(Strin
         .build();
   }
 
+  private boolean checkIfPastCompletedRecommendation(Recommendation recommendation,
+      DoctorsForDB doctor) {
+    final RecommendationGmcOutcome outcome = recommendation.getOutcome();
+    final boolean completed = outcome != null
+        && (APPROVED.getOutcome().equals(outcome.getOutcome())
+        || REJECTED.getOutcome().equals(outcome.getOutcome()));
+    //if doctor is under notice but has past completed revalidation -> start new recommendation
+    return doctor.getUnderNotice().equals(YES) && completed;
```

Comment:
Question rather than a comment... is this definitely the right field to use rather than `RecommendationStatus`?  I'm noticing how similar it would be to using a `new InProgressPredicate(null)`.

Also... because 2 enums are being used, you could do `return doctor.getUnderNotice().equals(YES) && (APPROVED.equals(outcome) || REJECTED.equals(outcome);`

Reply:
I would argue that as status is derived from outcome*****, outcome, it's safer/more correct to test against outcome than status.

Also the predicate test:
`return SUBMITTED_TO_GMC != r.getRecommendationStatus() || UNDER_REVIEW == r.getOutcome();`
returns true for NOT_STARTED and DRAFT as well as APPROVED/REJECTED, so don't think it can be used here

*****_it arguably shouldn't be, but that's the current implementation_